.github/workflows/release.yml

```diff
@@ -4,46 +4,197 @@ on:
+          (
+              cd build
+              ../source/configure
+              make doc/guide/html/index.html
+          )
 
-          git config user.name "GitHub Workflow"
-          git config user.email "cockpituous@cockpit-project.org"
+      - name: Update the website
+        run: |
+          rm -rf website/guide/latest
+          mv -Tv build/doc/guide/html website/guide/latest
 
-          echo '${{ secrets.GITHUB_TOKEN }}' > ~/.config/github-token
+          # Add frontmatter for Jekyll
+          find website/guide/latest -name '*.html' -exec sed -i '
+          1i\
+          ---\
+          layout: guide\
+          ---' '{}' ';'
 
-          # deploy key for updating the website repo, via tools/release-guide
-          eval $(ssh-agent)
-          ssh-add - <<< '${{ secrets.WEBSITE_DEPLOY_KEY }}'
+          git config --global user.name "GitHub Workflow"
+          git config --global user.email "cockpituous@cockpit-project.org"
+
+          cd website
+          git add guide/
+          git commit --message='Update guide to version ${{ needs.source.outputs.version }}'
+          git show --stat
+          git push origin main
+
+
+  flathub:
+    needs: source
+    environment: flathub
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source repository
+        uses: actions/checkout@v3
+        with:
+          path: src
 
-      - name: Run cockpituous
+      - name: Checkout flathub repository
+        uses: actions/checkout@v3
+        with:
+          path: flathub
+          repository: flathub/org.cockpit_project.CockpitClient
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+          # this is needed so we can push to a different repository
+          fetch-depth: 0
+
+      - name: Update flathub repository
         run: |
-          # override GitHub's bind mount from host, we don't want anything from there and it interferes with ssh
-          export HOME=$(getent passwd $(id -u) | cut -f6 -d:)
-          cd /build
-          release-runner -r https://github.com/$GITHUB_REPOSITORY -t $(basename $GITHUB_REF) tools/cockpituous-release
+          set -x
+
+          DOWNLOAD='${{ needs.source.outputs.download }}'
+          CHECKSUM='${{ needs.source.outputs.checksum }}'
+          VERSION='${{ needs.source.outputs.version }}'
+
+          git config --global user.name "GitHub Workflow"
+          git config --global user.email "cockpituous@cockpit-project.org"
+
+          cd flathub
+          git checkout -b "${VERSION}"
+          git add "$(../src/containers/flatpak/prepare "${DOWNLOAD}" "${CHECKSUM}")"
+          git commit -m "Update to version ${VERSION}"
+          git show
+          git push git@github.com:cockpit-project/org.cockpit_project.CockpitClient HEAD
+
 
   release-node-cache:
+    # doesn't depend on these, but let's make sure the build passes before we do this
+    needs: [source, flathub, guide]
```

Comment:
Bikeshedding, but IMHO this should only depend on `source`. Once the github release is done, the cache should be tagged as well. flathub and guide are follow-up actions, even if they fail, we should have a consistent github release.

Reply:
Changed.

I actually have an idea to draft the release, do all of the other work, and then finalize the release once it all succeeds.  That would be a nice follow-up.